lib/mix/tasks/test/watch.ex

```diff
@@ -48,9 +52,31 @@ defmodule Mix.Tasks.Test.Watch do
 
   defp run_tests(args) do
     IO.puts "\nRunning tests..."
-    :ok = args |> M.Command.build |> M.Command.exec
+    status = args |> M.Command.build |> M.Command.exec
     flush
-    :ok
+    status
+  end
+
+  @spec run_single_test(String.t, String.t) :: :ok
+
+  defp run_single_test(path, args) do
```

Comment:
This method is rather large and hard to read, perhaps break it down into multiple functions. Possibly a module?